modtools/reaction-product-trigger.lua

```diff
@@ -79,7 +79,11 @@ end
 
 local function afterProduce(reaction,reaction_product,unit,input_items,input_reagents,output_items)
  --printall(unit.job.current_job)
- local _,buildingId = dfhack.script_environment('modtools/reaction-trigger').getWorkerAndBuilding(unit.job.current_job)
+ local _,buildingId = -1,-1
+ -- unit.job.current_job might be nil for Adventure mode
+ if unit.job.current_job then
+  _,buildingId = dfhack.script_environment('modtools/reaction-trigger').getWorkerAndBuilding(unit.job.current_job)
+ end
  for _,hook in ipairs(productHooks[reaction.code] or {}) do
   local command = hook.command
   local processed = processArgs(command, reaction, reaction_product, unit, input_items, input_reagents, output_items, buildingId)
```

Comment:
@rndmvar specifically, I'm not sure that this line or the line after will work well if buildingId is -1 - it might pass through a building ID of -1 to callbacks that aren't expecting it. Would it be better to bail out in that case, similar to https://github.com/DFHack/scripts/blob/b330e696e6cb6a607d8cc77679bbe0fa5e51598c/modtools/reaction-trigger.lua#L134-L140 ? I'm not very familiar with modtools scripts.